.github/workflows/sage-test.yml

```diff
@@ -1,4 +1,7 @@
 name: Sage Install Tests
+permissions:
+  pull-requests: read
+  contents: read
```

Comment:
why these permissions needed?

Reply:
because we're pulling down the repository and doing a bunch of stuff on it. the tests failed to run without the permissions.